jobs/jobs/utils.py

```diff
@@ -585,7 +590,147 @@ async def get_job_progress(
         )
 
     response.update({"mode": str(job.mode)})
-    return response  # type: ignore
+    if job.mode == JobMode.Manual:
+        await update_manual_job_status(session, job, response)
+        return response
+
+    elif job.mode == JobMode.Automatic:
```

Comment:
Still not fixed. Please utilize if, no need for elif. If first if block is entered, it will exit the function with return statement